examples/step-27/step-27.cc

```diff
@@ -752,8 +718,8 @@ namespace Step27
         // We have to calculate the logarithms of absolute
         // values of coefficients and use it in linear regression fit to
         // obtain $\mu$.
-        for (unsigned int f = 0; f < res.second.size(); f++)
-          res.second[f] = std::log(res.second[f]);
+        for (double &f : res.second)
+          f = std::log(f);
```

Comment:
This is a bit hard to read, maybe, because the variable name `f` doesn't give away much. Maybe `residual_element`?